cmd/bpf2go/compile.go

```diff
@@ -67,6 +67,9 @@ func compile(args compileArgs) error {
 		"-g",
 	)
 	cmd.Dir = args.dir
+	cmd.Env = append(cmd.Env,
+		"BPF_CFLAGS='-D"+getTargetArch()+"'",
```

Comment:
You should pass `"-D"+getTargetArch()` directly to the compiler, not modify BPF_CFLAGS (which doesn't have any effect). I think the right place to add the flag is actually in main.go, where you can **prepend** it to `cFlags`. Prepending is important so that our users can override the define if they choose to do so.